libraries/Microsoft.Bot.Builder.Skills/Dialogs/SkillDialogArgs.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// A class with dialog arguments for a <see cref="SkillDialog"/>.
+    /// </summary>
+    public class SkillDialogArgs
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="BotFrameworkSkill"/> that the dialog will call.
+        /// </summary>
+        /// <value>
+        /// The <see cref="BotFrameworkSkill"/> that the dialog will call.
+        /// </value>
+        public BotFrameworkSkill Skill { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ActivityTypes"/> to send to the skill.
+        /// </summary>
+        /// <value>
+        /// The <see cref="ActivityTypes"/> to send to the skill.
+        /// </value>
+        public string ActivityType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the event or invoke activity to send to the skill (this value is ignored for other types of activities).
+        /// </summary>
+        /// <value>
+        /// The name of the event or invoke activity to send to the skill (this value is ignored for other types of activities).
+        /// </value>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value property for the activity to send to the skill.
+        /// </summary>
+        /// <value>
+        /// The value property for the activity to send to the skill.
+        /// </value>
+        public object Value { get; set; }
+
+        /// <summary>
+        /// Gets or sets the text property for the <see cref="ActivityTypes.Message"/> to send to the skill (ignored for other types of activities).
+        /// </summary>
+        /// <value>
+        /// The text property for the <see cref="ActivityTypes.Message"/> to send to the skill (ignored for other types of activities).
+        /// </value>
+        public string Text { get; set; }
```

Comment:
I'm not quite certain why these are broken out like this.  Is this to forward an activity to the skill?  If so, why not 
   public Activity Activity {get;set;}
That covers all bases.

Reply:
In begindialog we create a new activity and in continuedialog we forward what we've got in that turn. 
BeginDialog constructs the activity to start the skill and applies some additional parameters in  ApplyParentActivityProperties. 

We could change this to receive an activity (that the dev will have to create) and validate the type before forwarding but would like to hear @johnataylor feedback too.

Reply:
I think passing in an activity to begin makes a lot of sense.

Reply:
Thanks, I refactored DialogArgs and SkillDialog to take activity and updated the MainDialog code to create the activity there.